### ISSUE TITLE
Apple Silicon Support

### ISSUE THREAD

**Issue by mike85** [laravel/homestead#1552]
Is it possible to use Homestead with Apple Silicon already?

**Comment by svpernova09** [comment]
Not until Vagrant, and your provider of preference support the new chip & OS.

**Comment by mike85** [comment]
> Not until Vagrant, and your provider of preference support the new chip & OS.

If the providers will only support to virtualize ARM Linux images instead of x86, will settler also provide an ARM alternative to bento/ubuntu x86?

**Comment by svpernova09** [comment]
> 
> 
> > Not until Vagrant, and your provider of preference support the new chip & OS.
> 
> If the providers will only support to virtualize ARM Linux images instead of x86, will settler also provide an ARM alternative to bento/ubuntu x86?

I don't know if  Homestead will support ARM. The idea kinda terrifies me because I'm already managing 2 major LTS versions across 4 providers. Adding support for Apple Silicon (Assuming Vagrant & Providers get their stuff sorted out) would be a major task in Homestead.

I don't think I would build ARM support into Homestead as an open-source project. I believe the effort involved would be substantial and would require compensation. I don't even have a machine to test this on.

**Comment by mike85** [comment]
> > > Not until Vagrant, and your provider of preference support the new chip & OS.
> > 
> > 
> > If the providers will only support to virtualize ARM Linux images instead of x86, will settler also provide an ARM alternative to bento/ubuntu x86?
> 
> I don't know if Homestead will support ARM. The idea kinda terrifies me because I'm already managing 2 major LTS versions across 4 providers. Adding support for Apple Silicon (Assuming Vagrant & Providers get their stuff sorted out) would be a major task in Homestead.
> 
> I don't think I would build ARM support into Homestead as an open-source project. I believe the effort involved would be substantial and would require compensation. I don't even have a machine to test this on.

Thanks for the comment! Hopefully it will be possible to run x86 images on VMWare/Parallels/… one day.

**Comment by svpernova09** [comment]
https://twitter.com/JoePFerguson/status/1331601928309051400

**Comment by svpernova09** [comment]
https://joeferguson.me/homestead-on-apple-silicon/

**Comment by asterism612** [comment]
"homestead" because your Mac is equipped with the Apple M1 chip that does not support Intel-based operating systems. To create a compatible virtual machine, use an ISO or VHDX image file with an ARM-based operating system.

**Comment by svpernova09** [comment]
> 
> 
> "homestead" because your Mac is equipped with the Apple M1 chip that does not support Intel-based operating systems. To create a compatible virtual machine, use an ISO or VHDX image file with an ARM-based operating system.

https://joeferguson.me/homestead-on-apple-silicon/

**Comment by Fossil01** [comment]
So since M1 won't be supported anytime soon and I want to get net next 16 inch M1/M2 Mac when it comes out... I am looking at alternatives already :-)

Has anyone come up with an alternative neat and clean solution yet? I thought about just setting up a VM on my server and then using something like VSCode's Remote Code Development plugin (maybe over a VPN even) but a local solution with most stuff pre-configured in Vagrant is still pretty damn nice :-)

I don't like running in Docker with something like Sail as I have some custom applications I need, which I currently install with an `after.sh` script.

**Comment by dafvid** [comment]
Valet works just fine on M1. Unless your coordinating with Windows users I'd go with Valet instead.Sent from my Galaxy
-------- Original message --------From: Fossil ***@***.***> Date: 23/06/2021  10:13  (GMT+01:00) To: laravel/homestead ***@***.***> Cc: David Stenwall ***@***.***>, Manual ***@***.***> Subject: Re: [laravel/homestead] Apple Silicon Support (#1552) 
So since M1 won't be supported anytime soon and I want to get net next 16 inch M1/M2 Mac when it comes out... I am looking at alternatives already :-)
Has anyone come up with an alternative neat and clean solution yet? I thought about just setting up a VM on my server and then using something like VSCode's Remote Code Development plugin (maybe over a VPN even) but a local solution with most stuff pre-configured in Vagrant is still pretty damn nice :-)
I don't like running in Docker with something like Sail as I have some custom applications I need, which I currently install with an after.sh script.

**Comment by Fossil01** [comment]
I need Linux to properly develop some stuff, hence my use of Homestead :-)

**Comment by svpernova09** [comment]
For those that may be unaware, what "Homestead on M1 macOS" *really* encompasses are the following tasks:

* Build a Packer template for Ubuntu 20.04 on ARM. (Today we use chef/bento for x86_64, there is no ARM template)
* Test Settler's provision script on ARM Ubuntu 20.04.
* Test and configure Vagrant packaging with Parallels.

Then we'd have a *beta* base box to test drive Homestead with. You'd be required to buy/use Parallels as they're currently the only of our supported providers that have shipped M1 support. I could build a packer template without an M1 system. But without it, the effort would be wasted.

I believe this would also need to be a new Vagrant base box. I don't *think* you can mix architectures in the same box.


## Fundraising

I appreciate the support in terms of doing a fundraiser. I have some feelings about doing such during a still ongoing pandemic, but it's a possible option. I was hoping there would be a Laravel agency or group that would get value out of sponsoring such a project, but there hasn't been any interest there. Then it's up to individual contributions, How do I do that while remaining transparent about where the money is going *and* ensuring I'm reporting the incoming correctly to the IRS?  I would gladly provide a 1099 contract to a company that wanted to do this.

When people come into this same thread where I'm asking for help, let alone the repo, and ask about / push other Homestead alternatives: that comes across to me the maintainer (me) as really crappy. You probably didn't intend it to, but it's easy to forget there's a human behind the project you're using. There are *many* humans behind Homestead, I'm just the one who bugged Taylor enough. Homestead *isn't* my full-time job. I have a full-time job + I'm involved in running several nonprofits. I'm one of the few people who maintain Laravel package(s) but don't work *for* Laravel. I just really love Vagrant and Virtualization.

**Comment by browner12** [comment]
Once my GME moons, I'm fully funding this project, Joe. 🌕  💰

**Comment by Fossil01** [comment]
Fully understand where you are coming from @svpernova09. Honestly the only reason I asked for "alternatives" is because some other issue stated that M1 support was definitely not coming.

Let us know if you do go forwards with this and are accepting donations for M1 support. I do not currently own one but will probably switch with the next Macbook Pro's and would gladly drop some $$$ to make this happen.

**Comment by dtliao0303** [comment]
Hi all,

I'm new to Homestead and Vagrant, and I'm not sure if it's okay to share this here, apologies in advance if it's not cool. 

After a lot of googling I stumble upon this Homestead fork build for Apple Silicon, and works (PHP & MySQL) for me with almost no tweaking needed.

Note: it uses Parallel as the provider.
[yodoware/homestead-arm64](https://github.com/yodoware/homestead-arm64)

I hope this helps people out who are having trouble to run Homestead on their M1 Macs, and save them lots of time. Cheers.

**Comment by TjoBiZ** [comment]
I had installed but for me it doesn't work.
Didn't mount folders from parent MacOS 
Didn't up Appache with configs sites, not anyone site, only  one default apache config.
I see only default  Ubuntu 20.04.2 LTS (GNU/Linux 5.4.0-73-generic aarch64) version but I can do that install without this image.
 I have no idea what I can do next.

**Comment by TjoBiZ** [comment]
Yes, good idea.
It's work for me too.
but also I delete row with type system from config.
 Thnx. 
Will be waiting VirtualBox

**Comment by adriandmitroca** [comment]
Is there any update on this, @svpernova09 or the idea just got dropped completely?

**Comment by svpernova09** [comment]
> 
> 
> Is there any update on this, @svpernova09 or the idea just got dropped completely?

This issue seems to ebb and flow in interest. I've outlined what I would need to make this happen and everyone keeps telling me to crowdfund it myself. I get that a random nerd on the internet wanting people to buy them an expensive computer in exchange for building something is a *big* ask (and additionally harsh in a pandemic). I assumed there were agencies and companies out there using Homestead and would want to support its development. Apparently, there aren't that many focused on ARM macs or they've all moved on from Homestead. I always feel compelled to disclose that I *don't* work for Laravel, I don't get paid to work on Homestead. I work on Homestead because I love Linux and I want PHP developers to have a solid local development experience.

Since September 29th, 2019 (when I started [tracking downloads](http://vcdt.joeferguson.me/)) the base box for Homestead has been downloaded 686,396 times. That means not only did someone clone the Homestead repo, but they also ran `bash init.sh` to set up the box, and then they ran `vagrant up` *at least* long enough that the box downloaded was counted in Hashicorp's S3 metrics (which is what I query to get download counts) According to GitHub's Insights, Homestead Traffic continues to see ~2,000+ unique cloners per 2 weeks or so:

![image](https://user-images.githubusercontent.com/967362/134021705-7dec564b-fae2-445b-a3f7-296a111e2d4f.png)

If we look at Packagist to see what packages out there are [depending on Homestead, sorted by downloads](https://packagist.org/packages/laravel/homestead/dependents?order_by=downloads) shows us a picture of a user who has included Homestead in their dependencies for Laravel packages. The next listings are for a WordPress framework and [phpBB](https://packagist.org/packages/phpbb/phpbb) itself requires `laravel/homestead: ~7.0` (Really old version, likely doesn't even work)

![image](https://user-images.githubusercontent.com/967362/134022843-6c2223f0-168f-4167-b953-bd41106645f3.png)

I'm wondering if the inflated GitHub insights are from CICD clones, there doesn't seem to be any distinction in the UI. I find it hard to believe anyone has wired up `vagrant up` as part of their CICD processing. I want to believe everyone downloading the base box is Human, otherwise, that is a lot of wasted bandwidth.

Based on this information it's hard to see "who" is using Homestead. There's a number of people using Homestead as the issues here are typically pretty active and there's decent traffic in the Homestead Discord Channel of people asking questions and such.

**Comment by ttmwebdesign** [comment]
Hello @svpernova09 - I have two M1 Macs and am willing to be a beta tester for you. I use Homestead on my 2019 iMac and have been wanting to get it operational on my M1s but haven't had luck on my own. If you have any branches ready to offer up for testing let me know what to pull, or hit me up when they're available.

**Comment by skylerkatz** [comment]
@ttmwebdesign The issue at the moment is that Joe does not own an M1 machine, so he has not worked on the feature. If someone is able to provide him an M1 machine I believe he would be willing to work on an M1 version, as he stated in a recent blog post.

**Comment by svpernova09** [comment]
> We've just sponsored you @svpernova09, so hopefully you'll be able to get to work on this soon!

Wow. I'm going to spend a few minutes picking my jaw up off the floor.

**Comment by plakhin** [comment]
I think this can be interesting:
https://ubuntu.com/blog/canonical-transforms-linux-on-mac
https://github.com/Fred78290/vagrant-multipass

I was using multipass on Windows and it's super nice!

**Comment by vesper8** [comment]
Shouldn't https://github.com/utmapp/UTM be a possible new alternative to Virtualbox for running Homestead on M1 for free?

I understand currently Parallels is the only option for Apple Silicon, but correct me if I'm wrong, UTM does the same thing and is free and open source, so it should be a perfect candidate for Homestead on M1?

**Comment by Frittenfred97** [comment]
Since end of 2022 Virtualbox has released support for M1/M2. Maybe there is a possibility to check this for those who don't want to use Parallels.

source : https://osxdaily.com/2022/10/22/you-can-now-run-virtualbox-on-apple-silicon-m1-m2/

**Comment by svpernova09** [comment]
> Since end of 2022 Virtualbox has released support for M1/M2. Maybe there is a possibility to check this for those who don't want to use Parallels.
> 
> source : https://osxdaily.com/2022/10/22/you-can-now-run-virtualbox-on-apple-silicon-m1-m2/

I'm sure this looks great to the user who wants to avoid paying Parallels. Virtualbox has been nothing but a source of bugs and issues that end users blame the Homestead project on and it's certainly taken a toll on me over the years.

ICYMI: I've just spent 2-3 months trying to get a successful Virtualbox base box built for Homestead and was just finally able to make it happen by giving up on Apple and Windows completely.

I'm certainly in *no hurry* to start supporting another Virtualbox provider from the same company on a platform that isn't interoperable at all. Meaning: I can't build Apple Silicon on X86 systems. Thankfully an *amazing* Sponsor solved that issue which is what allows me to build them today, but I'm still not eager to add *another* base box. Especially since we're test-driving libvirt, another possible stable provider we could publish boxes for.


## TL;DR

I'm in absolutely zero hurry to even install Virtualbox 7.0 on my M1 system out of fear of instability it will introduce into the `homestead-arm` parallels provider builds. Virtualbox 6.x was a bear of a provider because bugs would crop up in dot releases and then re-emerge again in later releases. That's not something I'm in a hurry to get back to.

**Comment by pathros** [comment]
@Frittenfred97 VirtualBox doesn't support 64bits machines. Hence, VirtualBox is useless on Mac Apple Silicon M1.
I have just tried and it doesn't work with 64bits, just as shown [in this YT (Spanish Language) video](https://youtu.be/JaZQmTFCubM?t=736).

Regardless of using a paid software, such as Parallels, is there a (free software) way to install Laravel Homestead on a Mac Apple Silicon M1, from now on (February 2023)?

**Comment by browner12** [comment]
you could try the developer preview for Apple Silicon

https://download.virtualbox.org/virtualbox/7.0.6/VirtualBox-7.0.6_BETA4-155176-macOSArm64.dmg

**Comment by stebogit** [comment]
I believe this is still true, as my colleague just tried to setup Homestead and failed right away when trying to install Virtual Box on an M1 Mac.
Is the only option in this case to use Parallel? Is this going to ever change, or VB does not plan to have a working version for Apple's hardware?

If the latter is the case, is Laravel Sail the best alternative to Homestead? Or maybe something like [Lando](https://joeferguson.me/the-workshop-local-dev-with-lando-phparchitect-magazine-november-2022/) might be a more similar option (as in less Docker low level configuration)?

**Comment by svpernova09** [comment]
> I believe this is still true, as my colleague just tried to setup Homestead and failed right away when trying to install Virtual Box on an M1 Mac.
Is the only option in this case to use Parallel? Is this going to ever change, or VB does not plan to have a working version for Apple's hardware?

I don't have any current plans to support VirtualBox on Apple Silicon. VirtualBox has been a total nightmare to support on Linux & Windows and I'm in no hurry to even attempt it on macOS. That could change.

> If the latter is the case, is Laravel Sail the best alternative to Homestead? Or maybe something like [Lando](https://joeferguson.me/the-workshop-local-dev-with-lando-phparchitect-magazine-november-2022/) might be a more similar option (as in less Docker low level configuration)?

If you deploy to your server that runs `php artisan serve`, Sail is a great development environment for you. I don't deploy to this target so it doesn't make sense to develop against it.

Lando is a fantastic tool for local development that leverages Docker. I've been sponsoring them on GitHub until recently (which has nothing to do with my changing where those sponsorship dollars go). If you're using a Linux host or Windows WSL2 with Docker Desktop, you'll have a perfectly reasonable experience. On macOS, the fastest development environment is going to be Valet. 

Homestead on Parallels is going to be the most stable experience for Vagrant if you want a full Linux VM. I get not wanting to buy a Parallels license. Even more so now that VMWare has been sold off. 

Personally on Linux hosts I use Lando or my own configuration. On macOS hosts I use Valet, Lando, or Homestead (w/ Parallels). On Windows I reformat and install Fedora. Even WSL2 isn't enough to keep me developing on Windows. Windows 11 has had a lot to do with that.

**Comment by pathros** [comment]
I have found Laravel Herd a very nice alternative to, not only Laravel Homestead, but to Laravel Valet for Apple Silicon Mx.

Along with Laravel Herd you can use Docker for database support.

I used Laravel Homestead for many years, but since Apple Silicon I could no longer use it. Laravel Herd and Docker came to my rescue.

**Comment by stebogit** [comment]
Laravel Herd is not an isolated dev environment though, so if you have multiple projects using for instance different PHP versions you can't run them at the same time (I think)
